numpy/core/src/multiarray/scalarapi.c

```diff
@@ -234,6 +234,7 @@ PyArray_CastScalarToCtype(PyObject *scalar, void *ctypeptr,
     descr = PyArray_DescrFromScalar(scalar);
     castfunc = PyArray_GetCastFunc(descr, outcode->type_num);
     if (castfunc == NULL) {
+        Py_XDECREF(descr);
```

Comment:
Actually, this check should be done immediately, calling `PyArray_GetCastFunc` if `descr` is NULL is not going to work out.